framework/src/outputs/TopResidualDebugOutput.C

```diff
@@ -89,7 +90,8 @@ TopResidualDebugOutput::printTopResiduals(const NumericVector<Number> & residual
       for (const auto & dof : dof_indices)
         if (dof >= dof_map.first_dof() && dof < dof_map.end_dof())
         {
-          vec[j] = TopResidualDebugOutputTopResidualData(var_num, 0, residual(dof), true);
+          vec[j] =
+              TopResidualDebugOutputTopResidualData(var_num, {}, 0, Point(), residual(dof), true);
```

Comment:
Hmm... Now this call will cause a problem with the const ref to a set. I guess you can leave it as a copy unless you want to add another override.

Reply:
I didn't think I could store any constant references in the structure because of this no-argument constructor getting called in `vector<T>::resize`. As far as *passing* by constant reference, would I actually gain anything since the data immediately gets copied? I thought that the compiler would optimize out the double copy.

Reply:
Yeah, if this is your use case, then don't worry about the const ref. The compiler likely will optimize out the double copy and since this isn't in a loop. You never know thought, but we don't need to prematurely over-optimize it.